cmd/fleet/cron.go

```diff
@@ -359,6 +362,53 @@ func checkOvalVulnerabilities(
 	return results
 }
 
+func checkGovalDictionaryVulnerabilities(
+	ctx context.Context,
+	ds fleet.Datastore,
+	logger kitlog.Logger,
+	vulnPath string,
+	config *config.VulnerabilitiesConfig,
+	collectVulns bool,
+) []fleet.SoftwareVulnerability {
+	var results []fleet.SoftwareVulnerability
+
+	// Get Platforms
+	versions, err := ds.OSVersions(ctx, nil, nil, nil, nil)
+	if err != nil {
+		errHandler(ctx, logger, "listing platforms for goval_dictionary pulls", err)
+		return nil
+	}
+
+	if !config.DisableDataSync {
+		// Sync on disk goval_dictionary sqlite with current OS Versions.
+		downloaded, err := goval_dictionary.Refresh(ctx, versions, vulnPath)
+		if err != nil {
+			errHandler(ctx, logger, "updating goval_dictionary databases", err)
+		}
+		for _, d := range downloaded {
+			level.Debug(logger).Log("goval_dictionary-sync-downloaded", d)
+		}
+	}
+
+	// Analyze all supported os versions using the synced goval_dictionary definitions.
+	for _, version := range versions.OSVersions {
+		start := time.Now()
+		r, err := goval_dictionary.Analyze(ctx, ds, version, vulnPath, collectVulns)
+		elapsed := time.Since(start)
+		level.Debug(logger).Log(
```

Comment:
nit: this is an existing bug in OVAL processing as well where we are logging os versions not being scanned

Reply:
Roger that. I can probably tweak the is-supported check to happen further up the chain and stop the spurious logging for both. I definitely saw those logs heh

Reply:
Let me know if https://github.com/fleetdm/fleet/commit/ae836add9cb8cfbbbacb2376e884f142750d179f works for this. Updated godoc so the behavior I'm relying on is a bit less implicit. If we want to use a typed error here instead let me know.

Reply:
Actually, adding tweaks to log when we skip a platform since this is debug logging anyway.

Reply:
Actually actually, definitely need a typed error here because if I check for nil/nil then the empty array for the result matches nil on == and...that's not useful.

Reply:
Okay, that's fixed, and we now bounce an error back (and I think that error/the comparison should be cheap?) when a platform is unsupported.